### PR TITLE
feat: data connectors on the project page

### PR DESCRIPTION
**Pull request by ciyer**
/deploy renku-data-services=leafty/build-data-connectors

### CODE REVIEW

**Review by leafty — COMMENTED** [review]
Comments from manual testing:

1. When creating a new data connector from a project, the visibility is "private" by default, even on a public project.
Screenshot from 2024-10-09 13-49-51

2. When creating a data connector from a project page, using the "advanced mode" does not seem to work. I pasted the polybox config from `dev.renku.ch` but then the request sent to the backend only contained `type = webdav` and not the other fields.

File: client/src/features/dataConnectorsV2/components/DataConnectorView.tsx [hunk 1]
```diff
@@ -16,27 +16,96 @@
...
+
+function ConfigurationKeyIcon({ configKey }: { configKey: string }) {
+  if (configKey === "type") {
+    return <Database className={cx("bi", "me-1")} />;
+  }
+  if (configKey === "url" || configKey === "endpoint") {
+    return <Link45deg className={cx("bi", "me-1")} />;
+  }
+  if (configKey === "accessMode") {
+    return <UniversalAccessCircle className={cx("bi", "me-1")} />;
```
[the remark this answers is not in the document]
Comment (ciyer) [reply client/src/features/dataConnectorsV2/components/DataConnectorView.tsx:58]: There is a new design that does not use icons. The new design will be adopted before this feature is merged.

File: client/src/features/dataConnectorsV2/components/DataConnectorView.tsx [hunk 2]
```diff
@@ -16,27 +16,96 @@
...
+  }
+  if (configKey === "mountPoint") {
+    return <img src="/crosshair.svg" className={cx("bi", "me-1")} />;
```
[the remark this answers is not in the document]
Comment (ciyer) [reply client/src/features/dataConnectorsV2/components/DataConnectorView.tsx:61]: See above re: icons

File: client/public/crosshair.svg
[the remark this answers is not in the document]
Comment (ciyer) [reply client/public/crosshair.svg:1]: Fixed in 4e0d010c5799bd60a209a1c2103c466dc2288159

File: client/src/features/dataConnectorsV2/components/DataConnectorActions.tsx [hunk 3]
```diff
@@ -31,24 +32,84 @@ import {
...
-interface DataConnectorDeleteModalProps {
+interface DataConnectorRemoveModalProps {
   dataConnector: DataConnectorRead;
+  dataConnectorLink?: DataConnectorToProjectLink;
   isOpen: boolean;
   onDelete: () => void;
   toggleModal: () => void;
 }
-function DataConnectorDeleteModal({
+function DataConnectorRemoveModal({
```
[the remark this answers is not in the document]
Comment (ciyer) [reply client/src/features/dataConnectorsV2/components/DataConnectorActions.tsx:59]: Changed in 2b4eaabfc059d11c69985b136eab54063935202e

File: client/src/features/dataConnectorsV2/components/DataConnectorView.tsx [hunk 4]
```diff
@@ -16,27 +16,96 @@
...
+  DataConnectorRead,
+  DataConnectorToProjectLink,
+} from "../api/data-connectors.api";
+import { useGetDataConnectorsByDataConnectorIdSecretsQuery } from "../api/data-connectors.enhanced-api";
 import { storageSecretNameToFieldName } from "../../secrets/secrets.utils";
+
 import DataConnectorActions from "./DataConnectorActions";
+import useDataConnectorProjects from "./useDataConnectorProjects.hook";
+
+function ConfigurationKeyIcon({ configKey }: { configKey: string }) {
```
[the remark this answers is not in the document]
Comment (ciyer) [reply client/src/features/dataConnectorsV2/components/DataConnectorView.tsx:51]: We are going to remove all the icon stuff anyway, I would prefer to leave this as is in this PR and fix it in a later PR.

**Comment by ciyer** [comment]
> Comments from manual testing:
> 
> 1. When creating a new data connector from a project, the visibility is "private" by default, even on a public project.

Good point. I have fixed this in b1c73276bb2ee827a0a9dd43873d8bc87f3832cb.

> 2. When creating a data connector from a project page, using the "advanced mode" does not seem to work. I pasted the polybox config from `dev.renku.ch` but then the request sent to the backend only contained `type = webdav` and not the other fields.

I need to investigate this problem.

**Review by leafty — APPROVED** [review]
Great, thanks. Maybe we need to track the "advanced mode" issue.

**Comment by ciyer** [comment]
> Maybe we need to track the "advanced mode" issue.

I think this is the right way forward. I was able to create a data connector using advanced mode, so it does sometimes work. I have created an issue for this: https://github.com/SwissDataScienceCenter/renku-ui/issues/3349

**Pull request merged**